importer/trickle/trickledag.go

```diff
@@ -56,3 +56,181 @@ func fillTrickleRec(db *h.DagBuilderHelper, node *h.UnixfsNode, depth int) error
 	}
 	return nil
 }
+
+// TrickleAppend appends the data in `db` to the dag, using the Trickledag format
+func TrickleAppend(base *dag.Node, db *h.DagBuilderHelper) (*dag.Node, error) {
+	// Convert to unixfs node for working with easily
+	ufsn, err := h.NewUnixfsNodeFromDag(base)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get depth of this 'tree'
+	n, j := trickleDepthInfo(base, db.Maxlinks())
+	if n == 0 {
+		// If direct blocks not filled...
+		err := db.FillNodeLayer(ufsn)
+		if err != nil {
+			return nil, err
+		}
+		n++
```

Comment:
feels like a stray `n++` here, doesnt hurt to put it at the top of the if statement for readability.